qiskit/transpiler/passes/basis/basis_translator.py

```diff
@@ -294,17 +301,53 @@ def _replace_node(self, dag, node, instr_map):
                     node.op.params, node.op.name, target_params, target_dag
                 )
             )
-
         if node.op.params:
-            # Convert target to circ and back to assign_parameters, since
-            # DAGCircuits won't have a ParameterTable.
-            target_circuit = dag_to_circuit(target_dag)
-
-            target_circuit.assign_parameters(
-                dict(zip_longest(target_params, node.op.params)), inplace=True
-            )
+            parameter_map = dict(zip_longest(target_params, node.op.params))
+            bound_target_dag = target_dag.copy_empty_like()
+            for inner_node in target_dag.topological_op_nodes():
+                if any(isinstance(x, ParameterExpression) for x in inner_node.op.params):
+                    new_op = inner_node.op.copy()
+                    new_params = []
+                    for param in new_op.params:
+                        if not isinstance(param, ParameterExpression):
+                            new_params.append(param)
+                        else:
+                            bind_dict = {x: parameter_map[x] for x in param.parameters}
+                            if any(isinstance(x, ParameterExpression) for x in bind_dict.values()):
+                                new_value = param
+                                for x in bind_dict.items():
+                                    new_value = new_value.assign(*x)
+                            else:
+                                new_value = param.bind(bind_dict)
+                            if not new_value.parameters:
```

Comment:
Just to make our future-dev-life easier and remember why this is there 😛 
```suggestion
                            # cast from ParameterExpression to number, if no parameters left
                            if not new_value.parameters:
```